src/physics_swe.c

```diff
@@ -267,15 +267,16 @@ static PetscErrorCode RHSFunctionForInternalEdges(RDy rdy, Vec F, AMaxDiagnostic
       PetscReal hl = x_ptr[l * ndof + 0];
       PetscReal hr = x_ptr[r * ndof + 0];
 
-      if (amax_vec_int[ii] > amax_diags->max_wave_speed) {
-        amax_diags->max_wave_speed = amax_vec_int[ii];
-        amax_diags->global_edge_id = edges->global_ids[ii];
-      }
-
       if (!(hr < tiny_h && hl < tiny_h)) {
         PetscReal areal = cells->areas[l];
         PetscReal arear = cells->areas[r];
 
+        PetscReal cnum = amax_vec_int[ii] * edge_len / fmin(areal, arear) * rdy->dt;
```

Comment:
This is based on https://github.com/RDycore/toy-problems/blob/main/swe_roe/ex2b.c#L1670-1671